harvester/files/sources/anatomy_tool.py

```diff
@@ -0,0 +1,210 @@
+from collections import namedtuple
+from typing import Iterator
+
+import bs4
+from django.utils.text import slugify
+
+
+FileInfo = namedtuple("FileInfo", ["product", "mime_type", "url"])
+
+
+def get_file_infos(anatomy_tool_soup: bs4.BeautifulSoup) -> Iterator[FileInfo]:
+    for product in anatomy_tool_soup.find_all('record'):
+        mime_types = product.find_all('format')
+        urls = product.find_all('location')
+        if not urls:
+            continue
+        for mime_type, url in zip(mime_types, urls):
+            yield FileInfo(product, mime_type, url)
+
+
+class AnatomyToolFileExtraction(object):
+
+    youtube_regex = re.compile(r".*(youtube\.com|youtu\.be).*", re.IGNORECASE)
+    cc_url_regex = re.compile(r"^https?://creativecommons\.org/(?P<type>\w+)/(?P<license>[a-z\-]+)/(?P<version>\d\.\d)",
+                              re.IGNORECASE)
+    cc_code_regex = re.compile(r"^cc([ \-][a-z]{2})+$", re.IGNORECASE)
+
+    #############################
+    # OAI-PMH
+    #############################
+
+    @classmethod
+    def parse_copyright_description(cls, description: None | str) -> None | str:
+        if description is None:
+            return
+        elif description == "Public Domain":
+            return "pdm-10"
+        elif description == "Copyrighted":
+            return "yes"
+        url_match = cls.cc_url_regex.match(description)
+        if url_match is None:
+            code_match = cls.cc_code_regex.match(description)
+            return slugify(description.lower()) if code_match else None
+        license = url_match.group("license").lower()
+        if license == "mark":
+            license = "pdm"
+        elif license == "zero":
+            license = "cc0"
+        else:
+            license = "cc-" + license
+        return slugify(f"{license}-{url_match.group('version')}")
+
+    @classmethod
+    def get_oaipmh_external_id(cls, soup: bs4.BeautifulSoup, file_info: FileInfo) -> str:
+        el = file_info.product
+        return el.find('identifier').text.strip()
+
+    @classmethod
+    def get_set(cls, soup: bs4.BeautifulSoup, file_info: FileInfo) -> str:
+        return f"anatomy_tool:{file_info.product.find('identifier').text.strip()}"
+
+    #############################
+    # GENERIC
+    #############################
+
+    @staticmethod
+    def find_all_classification_blocks(element: bs4.element.Tag, classification_type: str, output_type: str)\
+            -> list[bs4.element.Tag]:
+        assert output_type in ["entry", "id"]
+        entries = element.find_all(string=classification_type)
+        blocks = []
+        for entry in entries:
+            classification_element = entry.find_parent('classification')
+            if not classification_element:
+                continue
+            blocks += classification_element.find_all(output_type)
+        return blocks
+
+    @classmethod
+    def get_files(cls, soup: bs4.BeautifulSoup, file_info: FileInfo) -> list[dict[str, str]]:
```

Comment:
Unused, can be deleted

Reply:
It's actually used in get_access_rights. Would you rather want me to refactor that method away or leave it as it is?